src/client/render/render-core.js

```diff
@@ -0,0 +1,350 @@
+import * as THREE from "three";
+import "three/examples/js/objects/Water";
+import "three/examples/js/objects/Sky";
+import "three/examples/js/loaders/LoaderSupport";
+import "three/examples/js/loaders/GLTFLoader";
+import * as Cookies from "js-cookie";
+import * as Stats from "stats-js";
+import * as config from "../config";
+import { CameraFlyControls } from "./cameras";
+import domReady from "../dom-ready";
+
+// From https://github.com/mrdoob/three.js/blob/master/examples/js/WebGL.js
+const _isWebGLAvailable = function() {
+	try {
+		const canvas = document.createElement("canvas");
+		return !!(window.WebGLRenderingContext && (canvas.getContext("webgl") || canvas.getContext("experimental-webgl")));
+	} catch (error) {
+		return false;
+	}
+};
+
+let _mainScene = null,
+	_renderer = null,
+	_marbles = {
+		meshes: []
+	},
+	animationUpdateFunctions = [],
```

Comment:
I'm in the dark on what you mean with this. Could you please expand on it? How would you envision this?

Reply:
The goal would be to have the functions that are called in the render loop clearly defined, so it's easier to see what happens during a render update. A simplified example would be this:
```js
function animate() {
  marbleManager?.updateMarbles(); // What this does and where the function comes from still unclear
  _controls.update(); // Might come from a Camera class later if we have tracking cams?
  activeLevel.update(); // Updates water and such
  _renderer.render();
}
```
Issue right now though, is that updating marbles is client-only (and depends on networking) which is done by adding it to the `animationUpdateFunctions` array somewhere. Once we have in-editor testing, those marbles need to be updated the same way but through physics.

We'd probably benefit from managing those marbles at the same level as renderCore (so it can call a function like above instead), but for that it needs some changes and planning ahead. How the marbles are updated internally (network or physics) would be handled in the manager and client/editor script(s).